### PR TITLE
scheduler: added default_irq_smp_affinity option

### PR DESCRIPTION
**Pull request by yarda**
The option 'default_irq_smp_affinity' affects what will be written to the
/proc/irq/default_smp_affinity. The option supports the following values:

calc:
  Content of the /proc/irq/default_smp_affinity will be calculated from
  the 'isolated_cores'. Non-isolated cores are calculated as an inversion
  of the 'isolated_cores'. Then the intersection of the non-isolated cores
  and the previous content of the /proc/irq/default_smp_affinity is
  written to the /proc/irq/default_smp_affinity. If the intersection is
  empty set, then just the non-isolated cores are written to the
  /proc/irq/default_smp_affinity. This behavior is the default if
  option 'default_irq_smp_affinity' is not specified.

ignore:
  The /proc/irq/default_smp_affinity is not touched by Tuned.

cpulist like e.g. '1,3-4':
  The cpulist is unpacked and directly written to the
  /proc/irq/default_smp_affinity

Examples:

[scheduler]
isolated_cores = 1, 2
default_irq_smp_affinity = calc

[scheduler]
isolated_cores = 1, 2
default_irq_smp_affinity = ignore

[scheduler]
isolated_cores = 1, 3
default_irq_smp_affinity = 2, 4

Resolves: rhbz#1896348

Signed-off-by: Jaroslav Škarvada <jskarvad@redhat.com>

### CODE REVIEW

**Review by marcel-apf — APPROVED** [review]
Looks good to me, thanks!

**Comment by jmencak** [comment]
I know the comment above and commit message could be considered a sort of a documentation, but really, it is not.  Can we start a practice of adding documentation to every new feature such as this directly in the doc/manual/... ?

**Comment by yarda** [comment]
> I know the comment above and commit message could be considered a sort of a documentation, but really, it is not. Can we start a practice of adding documentation to every new feature such as this directly in the doc/manual/... ?

It's good idea, I will try to transform the commit message into the asciidoc text.

**Comment by jmencak** [comment]
I've tested the default behaviour is unchanged + calc/ignore/cpulist seems to work too.  Rollback also works.  After the docs are added, happy to "lgtm".

**Comment by yarda** [comment]
An attempt to document this change.

After more thinking about it, I think this is not the best way to go. We should probably write such documentation into the code (e.g. by using docstrings) and then automatically generate structured reference guide from it. But it will require more man power to kick this in.

**Comment by yarda** [comment]
We can keep the whole asciidoc documentation (and further extend it), but I think that the chapter regarding the plugins and their options/parameters should be auto generated. This will allow easy maintenance.

File: doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc
Comment (jmencak) [inline doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc:90]: Suggestion:
s/what *Tuned* will write/the values *Tuned* writes/
s/to the/to/
Even though the file is "unique" I believe 'the' should be dropped as it is is "name".  A document person should review this.

**Comment by yarda** [comment]
Maybe there are still some 'wording' problems in the documentation added by this commit, waiting for feedback.

File: doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc
Comment (jmencak) [inline doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc:94]: s/of the/of/  (everywhere where you refer to *the* file :)

File: doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc
Comment (jmencak) [inline doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc:95]: s/of the `isolated_cores`/of `isolated_cores`/  I don't see isolated_cores mentioned anywhere, therefore I'd drop it here.

File: doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc
Comment (jmencak) [inline doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc:97]: s/is empty/is an empty/

File: doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc
Comment (jmencak) [inline doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc:100]: s/cpulist like e.g./for a cpulist such as/

**Review by jmencak — COMMENTED** [review]
Thanks for the docs!  Looks good, minor nits a documentation writer or a native English speaker should confirm.

**Comment by jmencak** [comment]
@courtneypacheco , could you please go over the docs?  Thank you!

File: doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc
Comment (yarda) [reply doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc:95]: It's mentioned on the previous line. I think the parameter `isolated_cores` should be described in its own paragraph preceding this text, but it's out of the context of this commit, so I didn't add it.

File: doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc
Comment (yarda) [reply doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc:100]: It's list describing 'calc', 'ignore', and cpulist parameters and it's introduced by the sentence: 'The following values are supported:', that's why I didn't want to write 'for a cpulist' but just 'cpulist'.

Comment (yarda) [reply doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc]: I reformatted the text to use asciidoc description, but I am not sure whether it looks better, nor whether the syntax I used is correct :)

**Comment by yarda** [comment]
I addressed most of the issues except those where I put comment.

File: doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc
Comment (jmencak) [inline doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc:103]: s/directly written/written directly/  ?

**Review by jmencak — APPROVED** [review]
Just a minor nit.  Looks good to me otherwise, thank you.  I've asked Courtney to review this as a native speaker of English.  You may want to give her a day or two before this merges.  Thank you. /cc @courtneypacheco

File: doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc
Comment (yarda) [reply doc/manual/modules/performance/ref_available-tuned-plug-ins.adoc:103]: I don't know :) but I changed it.

**Comment by jmencak** [comment]
The rendered asciidoc looks OK to me. Still /lgtm

**Comment by jmencak** [comment]
It is unlikely we are getting a native-speaker review of the docs within a reasonable timeframe.  I believe this is looks good and is good to go, @yarda

**Comment by yarda** [comment]
OK, thanks for info. I also think there is no point in blocking the merge due to it. We can update the text anytime later.

**Pull request merged**
